packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -412,9 +412,10 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
 
         if (this.activatedPlugins.get(pluginId)) {
             deferred.resolve();
+            return deferred.promise;
         }
         this.pluginActivationPromises.set(pluginId, deferred);
-        return deferred.promise;
+        return this.$activatePlugin(pluginId);
```

Comment:
`this.activatedPlugins` should be updated so if it is called twice you return it from there, right?

Reply:
it's being updated at startPlugin() method